tests/LinAlg/matrixTestsDenseRowMajor.cpp

```diff
@@ -156,6 +156,16 @@ const real_type* MatrixTestsDenseRowMajor::getLocalDataConst(hiop::hiopMatrixDen
   return amat->local_data_const();
 }
 
+/// Returns const pointer to local data block of matrix _A_.
+real_type* MatrixTestsDenseRowMajor::getLocalData(hiop::hiopMatrixDense* A)
+{
+  auto* amat = dynamic_cast<hiop::hiopMatrixDense*>(A);
+  if(amat == nullptr)
+    THROW_NULL_DEREF;
```

Comment:
nothing bad with this, but we haven't use exceptions in the core of hiop. I would just return null_ptr...

Reply:
I probably would keep this line, in order to keep consistence [with the other code in this file.](https://github.com/LLNL/hiop/blob/c559149f6cef1b71ebdea03cb56f820f29c2fde1/tests/LinAlg/matrixTestsRajaSparseTriplet.cpp)
`THROW_NULL_DEREF` is [defined here](https://github.com/LLNL/hiop/blob/c559149f6cef1b71ebdea03cb56f820f29c2fde1/tests/LinAlg/testBase.hpp#L58)

Reply:
We kept the exceptions out of hiop proper, but we found them to be really helpful when diagnosing bugs in our test suite and we probably won't miss the 2% performance hit. We can revert this, but I don't think we need to imo

Reply:
ok